### PR TITLE
set: add page (#2268)

### LINKED ISSUE #2268: page request: set
Requesting a page for the built-in command [set](http://linuxcommand.org/lc3_man_pages/seth.html)

### PR DESCRIPTION
**Pull request by mfrw**
Closes #2268 

----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

**Comment by agnivade** [comment]
Is this the only example common across all shells ?

**Comment by mfrw** [comment]
@agnivade no, there are more but I did not check across all available shells.
Can you suggest which shells should I consider?
I have strong feeling options should be the same for all, but I am not sure.

**Comment by agnivade** [comment]
Since it is in common, windows support is must. So I would say powershell, bash, zsh are essential.

**Comment by mfrw** [comment]
@agnivade if we have to include `powershell`  then I would suggest not adding anything else.

**Comment by agnivade** [comment]
You can also split the page into 2. Put one inside windows/ and the other with more options in common/. Windows clients will automatically take the more specific page.

**Comment by mfrw** [comment]
@agnivade I think this is a very good idea. Let me split them up into two pages :)
Thanks for the suggestion.

**Comment by mfrw** [comment]
@agnivade It turns out that `set` already had a windows page.

File: pages/common/set.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# set
+
+> Display, Set or Unset value of shell attributes and positional parameters.
```
Comment (sbrl) [inline pages/common/set.md]: Not sure that `Set` and `Unset` need to be capitalised here.

Comment (mfrw) [reply pages/common/set.md]: yes sure...

File: pages/common/set.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+`set -a`
+
+- Notify of job termination immediately:
+
+`set -b`
+
+- Set various options, e.g. enable `vi` style line editing:
+
+`set -o vi`
```
Comment (sbrl) [inline pages/common/set.md]: Perhaps `vi` should be surrounded by `{{` and `}}`?

File: pages/common/set.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# set
+
+> Display, set or unset value of shell attributes and positional parameters.
```
Comment (sbrl) [inline pages/common/set.md]: Perhaps `value` would read better as `values` here?

File: pages/common/set.md (hunk 2 above)
Comment (mfrw) [reply pages/common/set.md]: Ahhh... yes :(

File: pages/common/set.md (hunk 3 above)
Comment (sbrl) [reply pages/common/set.md]: Yeah, that would sound better than `... value of shell attributes ...`, as the `shell attributes` is plural, so the `value` should also be plural.

Comment (mfrw) [reply pages/common/set.md]: sure

Comment (mfrw) [reply pages/common/set.md]: @sbrl If you could help me in squashing all the commits as a single commit.
Just like gerrit does.

Comment (sbrl) [reply pages/common/set.md]: Yeah, when we merge a PR, we have a special button that lets us squash all the commits automagically - so you don't need to do anything there :slightly_smiling_face:

Comment (mfrw) [reply pages/common/set.md]: ahhh ! thanks

**Pull request merged**